File: 프로그래머스/unrated/150370. 개인정보 수집 유효기간/개인정보 수집 유효기간.cs
using System.Collections.Generic;
public class Solution
{
    public int Date(string date)
    {
        string[] dateArr = date.Split('.');
        return int.Parse(dateArr[0]) * 336 + int.Parse(dateArr[1]) * 28 + int.Parse(dateArr[2]);
    }

    public int[] solution(string today, string[] terms, string[] privacies)
    {
        Dictionary<char, int> term = new Dictionary<char, int>();

        foreach (string t in terms)
        {
            string[] temp = t.Split(' ');
            term.Add(char.Parse(temp[0]), int.Parse(temp[1]) * 28);
        }

        int td = Date(today);

        List<int> answer = new List<int>();

        for (int i = 0; i < privacies.Length; i++)
        {
            string[] temp = privacies[i].Split(' ');
            
            if (td >= Date(temp[0]) + term[char.Parse(temp[1])])
                answer.Add(i + 1);
                
        }

        return answer.ToArray();
    }
}

File: 프로그래머스/unrated/150370. 개인정보 수집 유효기간/개인정보 수집 유효기간.py
def Date(s):
    y, m, d = list(map(int, s.split('.')))
    return y*336 + m*28 + d

def Check(today, privacy, d):
    date, tp = privacy.split()
    date = Date(date)
    
    return today >= date + d[tp]

def solution(today, terms, privacies):
    today = Date(today)
    
    d = {term[0] : (int(term[2:]) * 28) for term in terms}
    
    return [i for i, v in enumerate(privacies, start=1) if Check(today, v, d)]